src/views/Alteration.vue

```diff
@@ -142,6 +155,18 @@ export default class Alteration extends Mixins(LegalApiMixin, FilingTemplateMixi
     Vue.nextTick(() => this.setHaveChanges(false))
   }
 
+  /** Fetch Business Snapshot */
+  private async fetchBusinessSnapshot (): Promise<any> {
+    return Promise.all([
+      this.getBusinessData(),
+      this.getBusinessData(BusinessDataTypes.TRANSLATIONS),
+      this.getBusinessData(BusinessDataTypes.ADDRESSES),
+      this.getBusinessData(BusinessDataTypes.DIRECTORS),
+      await this.getBusinessData(BusinessDataTypes.SHARE_CLASSSES),
+      await this.getContactInfo()
```

Comment:
I think you don't need the awaits here, since they already return a promise.

Also, you should add a check that you got an array with 6 responses, otherwise throw an error that something is missing.

Reply:
I recommend puttin the array length check in parseBusinessSnapshot() to:
- keep THIS method clean (just a return statement)
- protect parseBusinessSnapshot() from an out-of-bounds error if one of these fetches fails
- this would also clarify that parseBusinessSnapshot() expects an array of responses

Reply:
Sounds great! Added in the unit test commit 👍